cnd/src/seed.rs

```diff
@@ -12,82 +12,109 @@ use std::{
 };
 use thiserror;
 
-pub const SEED_LENGTH: usize = 32;
+const SEED_LENGTH: usize = 32;
+pub type Seed = [u8; SEED_LENGTH];
+
+/// We create a `RootSeed` either randomly or by reading in the PEM file from
+/// disk.  This `RootSeed` is used to generate a per swap `SwapSeed` which is
+/// then use as the secret source for deriving redeem/refund identities.
+/// `RootSeed` and `SwapSeed` are the same underlying type (`Seed`), they exist
+/// solely to allow the compiler to provide us with type safety.
+
+pub trait DeriveSwapSeed {
+    fn derive_swap_seed(&self, id: SwapId) -> SwapSeed;
+}
+
+impl DeriveSwapSeed for RootSeed {
+    fn derive_swap_seed(&self, id: SwapId) -> SwapSeed {
+        SwapSeed(self.sha256_with_seed(&[b"SWAP", id.0.as_bytes()]))
```

Comment:
If you make `Seed` an actual newtype, the `sha256_with_seed` function can live on there so you don't have to abuse `SwapSeed` here :)

Reply:
Whats the abuse?  Totally honest question, other suggestions implemented but this function is now 
```
impl DeriveSwapSeed for RootSeed {
    fn derive_swap_seed(&self, id: SwapId) -> SwapSeed {
        SwapSeed(Seed(self.sha256_with_seed(&[b"SWAP", id.0.as_bytes()])))
    }
}
```

Reply:
Don't mind me, I thought you are wrapping `self` inside `SwapSeed` to access a function. I can't read apparently.

Reply:
All g!